resources/js/components/Product/Price.vue

```diff
@@ -0,0 +1,124 @@
+<script>
+export default {
+    props: {
+        product: {
+            type: Object,
+            default: () => config.product || {},
+        },
+        location: {
+            type: String,
+            default: 'catalog',
+        },
+        options: {
+            type: Object,
+            default: () => ({}),
+        },
+    },
+
+    render() {
+        return this.$scopedSlots.default(this)
+    },
+
+    methods: {
+        calculatePrice(options = null) {
+            options ??= this.options
+            let product = options?.product || this.product
+            let location = options?.location || this.location
+
+            let special_price = options.special_price ?? false
+            let displayTax = this.$root.includeTaxAt(location)
+            let price = options?.price || (special_price ? product.special_price ?? product.price ?? 0 : product.price ?? 0) * 1
+
+            if (options.tier_price) {
+                price = options.tier_price.price * 1
+            }
+
+            if (options.product_options) {
+                price += this.calculateOptionsValue(price, product, options.product_options)
+            }
+
+            let taxMultiplier = this.getTaxPercent(product) + 1
+
+            return displayTax ? price * taxMultiplier : price / taxMultiplier
+        },
+
+        getTaxPercent(product) {
+            let country_id = window.config.tax.defaults.country_id
+            let region_id = window.config.tax.defaults.region_id
+            let postcode = window.config.tax.defaults.postcode
+
+            if (['shipping', 'billing'].includes(window.config?.tax?.calculation.based_on)) {
+                country_id = window.app?.$data?.checkout?.[window.config?.tax?.calculation.based_on + '_address']?.country_id || country_id
+                region_id = window.app?.$data?.checkout?.[window.config?.tax?.calculation.based_on + '_address']?.region_id || region_id
+                postcode = window.app?.$data?.checkout?.[window.config?.tax?.calculation.based_on + '_address']?.postcode || postcode
```

Comment:
```js
let basedOn = window.app?.$data?.checkout?.[window.config?.tax?.calculation.based_on + '_address']
```

And then use that instead of repeating it 3 times, would probably be much cleaner here.